Mathlib/Order/Basic.lean

```diff
@@ -955,21 +955,76 @@ theorem const_lt_const : const β a < const β b ↔ a < b := by simpa [Pi.lt_de
 
 end Function
 
-/-! ### Lifts of order instances -/
+/-! ### Pullbacks of order instances -/
+
+/-- Pull back a `Preorder` instance along an injective function.
+
+See note [reducible non-instances]. -/
+abbrev Function.Injective.preorder [Preorder β] [LE α] [LT α] (f : α → β)
+    (le : ∀ {x y}, f x ≤ f y ↔ x ≤ y) (lt : ∀ {x y}, f x < f y ↔ x < y) :
+    Preorder α where
+  le_refl _ := le.1 <| le_refl _
+  le_trans _ _ _ h₁ h₂ := le.1 <| le_trans (le.2 h₁) (le.2 h₂)
+  lt_iff_le_not_ge _ _ := by
+    rw [← le, ← le, ← lt, lt_iff_le_not_ge]
+
+/-- Pull back a `PartialOrder` instance along an injective function.
+
+See note [reducible non-instances]. -/
+abbrev Function.Injective.partialOrder [PartialOrder β] [LE α] [LT α] (f : α → β)
+    (hf : Function.Injective f)
+    (le : ∀ {x y}, f x ≤ f y ↔ x ≤ y) (lt : ∀ {x y}, f x < f y ↔ x < y) :
+    PartialOrder α where
+  __ := Function.Injective.preorder f le lt
+  le_antisymm _ _ h₁ h₂ := hf <| le_antisymm (le.2 h₁) (le.2 h₂)
+
+/-- Pull back a `LinearOrder` instance along an injective function.
+
+See note [reducible non-instances]. -/
+abbrev Function.Injective.linearOrder [LinearOrder β] [LE α] [LT α] [Max α] [Min α] [Ord α]
+    [DecidableEq α] [DecidableLE α] [DecidableLT α] (f : α → β)
+    (hf : Function.Injective f) (le : ∀ {x y}, f x ≤ f y ↔ x ≤ y) (lt : ∀ {x y}, f x < f y ↔ x < y)
+    (min : ∀ x y, f (x ⊓ y) = f x ⊓ f y) (max : ∀ x y, f (x ⊔ y) = f x ⊔ f y)
+    (compare : ∀ x y, compare (f x) (f y) = compare x y) :
+    LinearOrder α where
+  toPartialOrder := hf.partialOrder _ le lt
+  toDecidableLE := ‹_›
+  toDecidableEq := ‹_›
+  toDecidableLT := ‹_›
+  le_total _ _ := by simp only [← le, le_total]
+  min_def _ _ := by simp_rw [← hf.eq_iff, ← le, apply_ite f, ← min_def, min]
+  max_def _ _ := by simp_rw [← hf.eq_iff, ← le, apply_ite f, ← max_def, max]
+  compare_eq_compareOfLessAndEq _ _ := by
+    simp_rw [← compare, LinearOrder.compare_eq_compareOfLessAndEq, compareOfLessAndEq, ← lt,
+      hf.eq_iff]
+
+/-!
+### Lifts of order instances
+
+Unlike the constructions above, these construct new data fields.
+They should be avoided if the types already define any order or decidability instances.
+-/
 
 /-- Transfer a `Preorder` on `β` to a `Preorder` on `α` using a function `f : α → β`.
+
+See also `Function.Injective.preorder` when only the proof fields need to be transferred.
+
 See note [reducible non-instances]. -/
-abbrev Preorder.lift [Preorder β] (f : α → β) : Preorder α where
-  le x y := f x ≤ f y
-  le_refl _ := le_rfl
-  le_trans _ _ _ := _root_.le_trans
-  lt x y := f x < f y
-  lt_iff_le_not_ge _ _ := _root_.lt_iff_le_not_ge
+abbrev Preorder.lift [Preorder β] (f : α → β) : Preorder α :=
+  letI _instLE : LE α := ⟨fun a b ↦ f a ≤ f b⟩
+  letI _instLT : LT α := ⟨fun a b ↦ f a < f b⟩
+  Function.Injective.preorder f .rfl .rfl
```

Comment:
Oh no! We now have `Preorder.lift` and `Function.Injective.preorder` with have basically the same assumptions but different names? That sounds maximally confusing!

Are you claiming this is okay because there's only one of these and it matches all other ones? From a mathematical standpoint, it is expected that `Preorder` is different to other classes: it model-theoretically represents a *relational theory*, while most classes are instead algebraic.

Reply:
I claim the assumptions are quite different, one constructs data while the other does not.